File: python/1094.py
n = int(input())

total = 0
rabbitsTotal = 0
ratsTotal = 0
frogsTotal = 0

for i in range(n):
    qty, tp = input().split()
    qty = int(qty)
    
    if tp == 'C':
        rabbitsTotal += qty
    elif tp == 'R':
        ratsTotal += qty
    elif tp == 'S':
        frogsTotal += qty
        
    total += qty
    
print(f'Total: {total} cobaias')
print(f'Total de coelhos: {rabbitsTotal}')
print(f'Total de ratos: {ratsTotal}')
print(f'Total de sapos: {frogsTotal}')
print(f'Percentual de coelhos: {(rabbitsTotal * 100 / total):.2f} %')
print(f'Percentual de ratos: {(ratsTotal * 100 / total):.2f} %')
print(f'Percentual de sapos: {(frogsTotal * 100 / total):.2f} %')
    

File: lua/1094.lua
local n = tonumber(io.read())
local total = 0
local cTotal = 0
local rTotal = 0
local sTotal = 0
for i=1,n do
    local entry = io.read()
    local t = {}
    for v in string.gmatch(entry, '%S+') do
        table.insert(t, v)
    end
    local num = tonumber(t[1])
    local type = t[2]
    if type == 'C' then
        cTotal = cTotal + num
    elseif type == 'R' then
        rTotal = rTotal + num
    elseif type == 'S' then
        sTotal = sTotal + num
    end
    total = total + num
end

print(string.format('Total: %i cobaias', total))

print(string.format('Total de coelhos: %i', cTotal))
print(string.format('Total de ratos: %i', rTotal))
print(string.format('Total de sapos: %i', sTotal))

print(string.format('Percentual de coelhos: %.2f ' , cTotal * 100 / total)..'%')
print(string.format('Percentual de ratos: %.2f ', rTotal * 100 / total)..'%')
print(string.format('Percentual de sapos: %.2f ', sTotal * 100 / total)..'%')